src/connectors/snowflake/trulens/connectors/snowflake/connector.py

```diff
@@ -126,26 +124,22 @@ def _init_with_snowpark_session(
             )
         database_args["engine_params"]["paramstyle"] = "qmark"
 
-        required_settings = {
-            "account": snowpark_session.get_current_account(),
-            "user": snowpark_session.get_current_user(),
-            "database": snowpark_session.get_current_database(),
-            "schema": snowpark_session.get_current_schema(),
-            "warehouse": snowpark_session.get_current_warehouse(),
-            "role": snowpark_session.get_current_role(),
-        }
-        for k, v in required_settings.items():
-            if not v:
-                raise ValueError(f"`{k}` not set in `snowpark_session`!")
+        if connection_parameters is None:
+            connection_parameters = {
+                "account": snowpark_session.get_current_account(),
+                "user": snowpark_session.get_current_user(),
+                "database": snowpark_session.get_current_database(),
+                "schema": snowpark_session.get_current_schema(),
+                "warehouse": snowpark_session.get_current_warehouse(),
+                "role": snowpark_session.get_current_role(),
+                "password": "password",
+            }
+            for k, v in connection_parameters.items():
+                if not v:
+                    raise ValueError(f"`{k}` not set in `snowpark_session`!")
 
         database_url = URL(
-            account=snowpark_session.get_current_account(),
-            user=snowpark_session.get_current_user(),
-            password="password",
-            database=snowpark_session.get_current_database(),
-            schema=snowpark_session.get_current_schema(),
-            warehouse=snowpark_session.get_current_warehouse(),
-            role=snowpark_session.get_current_role(),
+            **connection_parameters,
```

Comment:
won't we still have the same problem with the dashboard if ppl use a `snowflake_session` to connect?

Reply:
yea, I'll add a warning in that case

Reply:
we'd need it for server-side evals. In the warehouse env, we only have a snowpark session

Reply:
FYI: this isn't technically correct as you can use the DB url to do it. That's how the stored proc works right now as we don't even use `SnowflakeConnector` and just use the DB url. Arguably, this should be changed to use the `SnowflakeConnector` though. I'm not sure as I haven't thought what's best.